File: Backend/models/exercise_log.py
from sqlalchemy import Column, Integer, Float, ForeignKey, DateTime
from sqlalchemy.orm import relationship
from datetime import datetime

from database.database import Base

class ExerciseLog(Base):
    __tablename__ = "exercise_logs"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    exercise_id = Column(Integer, ForeignKey("exercises.id"), nullable=False)
    duration_minutes = Column(Float)
    date = Column(DateTime, default=datetime.utcnow)

    # Relationships
    user = relationship("User", back_populates="exercise_entries")
    exercise = relationship("Exercise")

File: Frontend/lib/models/exercise_log.dart
class ExerciseLog {
  final int id;
  final int userId;
  final int exerciseId;
  final double durationMinutes;
  final DateTime date;

  ExerciseLog({
    required this.id,
    required this.userId,
    required this.exerciseId,
    required this.durationMinutes,
    required this.date,
  });
}